### Stress level prediction

In [1]:
import pandas as pd
import numpy as nd

In [2]:
ds_original=pd.read_csv("/workspaces/Stress_detection/data/stress_detection_data.csv")
print(ds_original.info())

shape=ds_original.shape
print(f"Contamos con  {shape[0]}  filas (individuos) y {shape[1]} columnas")

duplicados = ds_original.duplicated()
num_duplicados = duplicados.sum()
print(f"Existen {num_duplicados} datos duplicados")

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 773 entries, 0 to 772
Data columns (total 22 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Age                  773 non-null    int64  
 1   Gender               773 non-null    object 
 2   Occupation           773 non-null    object 
 3   Marital_Status       773 non-null    object 
 4   Sleep_Duration       773 non-null    float64
 5   Sleep_Quality        773 non-null    float64
 6   Wake_Up_Time         773 non-null    object 
 7   Bed_Time             773 non-null    object 
 8   Physical_Activity    773 non-null    float64
 9   Screen_Time          773 non-null    float64
 10  Caffeine_Intake      773 non-null    int64  
 11  Alcohol_Intake       773 non-null    int64  
 12  Smoking_Habit        773 non-null    object 
 13  Work_Hours           773 non-null    int64  
 14  Travel_Time          773 non-null    float64
 15  Social_Interactions  773 non-null    int

Contamos con 22 columnas y 773 filas.

No contamos con datos duplicados.

No contamos con datos nulos en ninguna variable.

### Datos numéricos

- Age: The age of the individual.
- Sleep_Duration: The total amount of sleep an individual gets on average (in hours).
- Sleep_Quality: Self-reported quality of sleep on a scale (e.g., 1-5).
- Physical_Activity: Frequency of physical activity (e.g., number of days/week or a score).
- Screen_Time: The average number of hours spent on electronic screens per day.
- Caffeine_Intake: The average amount of caffeine consumed (e.g., cups of coffee).
- Alcohol_Intake: The average amount of alcohol consumed (e.g., number of drinks per week).
- Work_Hours: The number of hours worked per week.
- Travel_Time: The time spent commuting (e.g., in hours per day).
- Social_Interactions: The frequency of social interactions (e.g., number of social events attended).
- Blood_Pressure: The individual's blood pressure level (e.g., 120/80).
- Cholesterol_Level: The individual's cholesterol level (e.g., in mg/dL).
- Blood_Sugar_Level: The individual's blood sugar level (e.g., in mg/dL).

### Datos categóricos:

- Gender: Gender of the individual (Male/Female).
- Occupation: The profession or job title of the individual.
- Marital_Status: The marital status of the individual (e.g., Married, Single).
- Wake_Up_Time: The time the individual wakes up on average.
- Bed_Time: The time the individual goes to bed on average.
- Smoking_Habit: Whether the individual smokes (Yes/No).
- Meditation_Practice: Whether the individual practices meditation (Yes/No).
- Exercise_Type: The type of exercise the individual engages in (e.g., Yoga, Cardio, Strength Training).
- Stress_Detection: A classification or indicator of the individual's stress level (e.g., High, Medium, Low).

## Análisis Exploratorio de Datos